### PR TITLE
WT-10251 Some C++ Workgen improvement

### PR DESCRIPTION
**Pull request by etienneptl**
Those are improvements I thought could be worth making while investigating the Workgen code:

- Declare variables where they are used
- Use `std::string` instead of `const char *`
- Use `static_cast` instead of C type casting
- Use `nullptr` instead of `NULL`
- Declare default constructors and destructors with the `default` keyword when empty

There are other areas to improve but that's a suggested start.

### CODE REVIEW

**Review by ddanderson — APPROVED** [review]
As long as SWIG is happy with the new changes, they all look good to me.

File: bench/workgen/workgen.cxx [hunk 1]
```diff
@@ -189,39 +187,43 @@ int
...
-
-    cycle_count = 0;
-    if ((ret = conn->open_session(conn, NULL, NULL, &session)) != 0) {
-        THROW("Error Opening a Session.");
+    if (conn->open_session(conn, nullptr, nullptr, &session) != 0) {
+        THROW("Error opening a session.");
     }
 
-    for (cycle_count = 0; !stopping; ++cycle_count) {
+    int ret;
```
Comment (jeremythorp) [inline bench/workgen/workgen.cxx:194]: I would suggest moving 'int ret' into the loop, and initialising it, as well as moving 'uri' into the loop too.

**Review by jeremythorp — APPROVED** [review]
Thanks Etienne, 
I think those changes all make sense and are valid improvements.
As a potential future improvement, I think the factory code in `Operation::init_internal()` could be changed to avoid requiring so many static_casts  and thereby becoming more OO. `OperationInternal *_internal` would likely be better as a smart pointer too.
Cheers, Jeremy

File: bench/workgen/workgen.cxx (hunk 1 above)
Comment (etienneptl) [reply bench/workgen/workgen.cxx]: Good suggestion, fixed in 35adee12db6765de98568b6b01b91f78bddeed2e

**Pull request merged**